packages/nouns-contracts/contracts/NounsArt.sol

```diff
@@ -0,0 +1,453 @@
+// SPDX-License-Identifier: GPL-3.0
+
+/// @title The Nouns art storage contract
+
+/*********************************
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░█████████░░█████████░░░ *
+ * ░░░░░░██░░░████░░██░░░████░░░ *
+ * ░░██████░░░████████░░░████░░░ *
+ * ░░██░░██░░░████░░██░░░████░░░ *
+ * ░░██░░██░░░████░░██░░░████░░░ *
+ * ░░░░░░█████████░░█████████░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ *********************************/
+
+pragma solidity ^0.8.6;
+
+import { INounsArt } from './interfaces/INounsArt.sol';
+import { Inflate } from './libs/Inflate.sol';
+import { SSTORE2 } from './libs/SSTORE2.sol';
+
+contract NounsArt is INounsArt {
+    // Current Nouns Descriptor address
+    address public override descriptor;
+
+    // Pending Nouns Descriptor address
+    address public override pendingDescriptor;
+
+    // Noun Backgrounds (Hex Colors)
+    string[] public override backgrounds;
+
+    // Noun Color Palettes (Index => Hex Colors)
+    mapping(uint8 => address) public palettesPointers;
+
+    // Noun Bodies (Custom RLE)
+    Trait public _bodies;
+
+    // Noun Accessories (Custom RLE)
+    Trait public _accessories;
+
+    // Noun Heads (Custom RLE)
+    Trait public _heads;
+
+    // Noun Glasses (Custom RLE)
+    Trait public _glasses;
+
+    /**
+     * @notice Require that the sender is the descriptor.
+     */
+    modifier onlyDescriptor() {
+        if (msg.sender != descriptor) {
+            revert SenderIsNotDescriptor();
+        }
+        _;
+    }
+
+    constructor(address _descriptor) {
+        descriptor = _descriptor;
```

Comment:
nit: solidity style guide says to use trailing `_`. I know it's mixed in the throughout the repo but maybe better to choose one

Reply:
yeah I like trailing myself, but stuck with this to be more consistent with the rest of the repo. I like repo consistency. if you still think it's better to change this I am down. wdyt?